tf_quant_finance/math/jacobian.py

```diff
@@ -0,0 +1,120 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for computing jacobian."""
+
+
+import functools
+
+import tensorflow.compat.v2 as tf
+
+
+def jacobian(func, xs, unconnected_gradients=None, name=None,
+             parallel_iterations=None, experimental_use_pfor=True):
+  """Computes the jacobian of `func` wrt to `xs`.
+  Args:
+    func: Python callable accepting one `Tensor` of shape of `x` and returning
+      a `Tensor` of any shape. The function whose jacobian is to be computed.
+    xs: A `Tensor` of parameters of `f` for which to differentiate. The
+      dimension (ndim) should be 2, and the first dimension is the batch
```

Comment:
I think `xs` can be a list of `Tensor`s as well. How about

xs: Python list of parameters of `f` for which to differentiate. (Can also be single `Tensor`). Each element of `xs` should be a float `Tensor` of either rank 1 or 2. If rank 2, then the first dimension is treated as a batch dimension. If the rank is 1, the batch dimension is assumed to be `[1]`

Reply:
Also, could you please check, when xs is a list if all the entries should have the same batch dimension? I.e.,
Can I have xs[0] of shape [3] and xs[1] of shape [2, 4]? Or should it be xs[0] of shape [2, 3]?

Reply:
As discussed before, addressed in 8a357ce with testing the multivariate quadratic function.